src/test/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibraryUnitTest.java

```diff
@@ -19,9 +18,170 @@
 import static org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary.*;
 
 /**
- * Tests for the Wellformed read filter. .
+ * Tests for the read filter library.
  */
-public final class WellformedReadFilterUnitTest extends ReadFilterTest {
+public final class ReadFilterLibraryUnitTest {
+    private static final int DEFAULT_READ_GROUP_COUNT = 5;
+    private static final String DEFAULT_READ_GROUP_PREFIX = "ReadGroup";
+    private static final String DEFAULT_PLATFORM_UNIT_PREFIX = "Lane";
+    private static final String DEFAULT_PLATFORM_PREFIX = "Platform";
+    private static final int DEFAULT_CHROMOSOME_COUNT = 1;
+    private static final int DEFAULT_CHROMOSOME_START_INDEX = 1;
+    private static final int DEFAULT_CHROMOSOME_SIZE = 1000;
+
+    private final int groupCount;
+    private SAMFileHeader header;
+
+    /**
+     * Construct a read filter test with the default number of groups
+     *  ({@link #DEFAULT_READ_GROUP_COUNT}.
+     */
+    @SuppressWarnings("unused")
+    public ReadFilterLibraryUnitTest() {
+        this.groupCount = DEFAULT_READ_GROUP_COUNT;
+    }
+
+    /**
+     * Returns the mock-up SAM file header for testing.
+     *
+     * @throws IllegalStateException if the header was not initialized
+     *          invoking {@link #beforeClass()}
+     * @return never <code>null</code>
+     */
+    private SAMFileHeader getHeader() {
+        if (header == null) {
+            throw new IllegalArgumentException(
+                    "header has not been initialized;"
+                            + " beforeClass() was not invoked");
+        }
+        return header;
+    }
+
+    /**
+     * Composes the Id for the read group given its index.
+     *
+     * This methods must return a unique distinct ID for each possible index and
+     * it must be the same value each time it is invoked.
+     *
+     * @param index the index of the targeted read group in the range
+     *              [1, {@link #groupCount}]
```

Comment:
It seems like we should change the indexing so that it's from [0, groupCount) and then just adjust up in the places where it actually matters.  Having an atypical indexing scheme with checks everywhere seems strange.